audiofile/core/io.py

```diff
@@ -144,9 +168,20 @@ def read(
         >>> signal, sampling_rate = read('stereo.wav', duration=0.1)
         >>> signal.shape
         (2, 800)
+        >>> signal, sampling_rate = read('stereo.wav', duration='400')
+        >>> signal.shape
+        (2, 400)
 
     """
     file = audeer.safe_path(file)
+
+    if duration is not None or offset != 0:
```

Comment:
Maybe better: `offset > 0` or what happens if offset is negative?

Reply:
Here we need to test `!=0` as `0` is the default value.

But we could extend the function to automatically do something like
```python
duration = max(duration, 0)
offset = max(offset, 0)
```
after they are converted to seconds.

Reply:
At the moment the behavior is not really predictable, e.g.:


```python
_, axs = plt.subplots(3, 1)

x, sr = af.read('test.wav')
audplot.waveform(x, ax=axs[0])

x, sr = af.read('test.wav', duration=0.5)
audplot.waveform(x, ax=axs[1])

x, sr = af.read('test.wav', duration=-0.5)
audplot.waveform(x, ax=axs[2])

plt.show()
```

![image](https://user-images.githubusercontent.com/10383417/217464601-780309e0-7641-441f-aba2-73c26b18026b.png)

but when we change from -0.5 to -0.1:

```python
_, axs = plt.subplots(3, 1)

x, sr = af.read('test.wav')
audplot.waveform(x, ax=axs[0])

x, sr = af.read('test.wav', duration=0.1)
audplot.waveform(x, ax=axs[1])

x, sr = af.read('test.wav', duration=-0.1)
audplot.waveform(x, ax=axs[2])

plt.show()
```

![image](https://user-images.githubusercontent.com/10383417/217464711-57971620-53c3-4235-a7cf-55b2e332c5b7.png)

But maybe instead of setting the values to 0 we could also interpret them from the end of the file? I.e. `duration=-0.5` crops off the last 0.5 seconds. And `offset=-0.5` returns the last 0.5 seconds.

Reply:
You are right. I created https://github.com/audeering/audiofile/issues/115 to handle this independent of this pull request.